air/src/stack/stack_flag/mod.rs

```diff
@@ -0,0 +1,801 @@
+        .iter_mut()
+        .take(64)
+        .skip(32)
+        .for_each(|v| *v *= not0_1);
+
+    degree_7_intermediate_values
+}
+
+/// This helper function generates all the flag values of degree seven and degree four
+/// stack operations.
+fn set_op_flags_degree_4_and_6<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+) -> ([E; NUM_DEGREE_6_OPS], [E; NUM_DEGREE_4_OPS]) {
+    let mut degree_6_intermediate_values = [E::ZERO; NUM_DEGREE_6_OPS];
+    let mut degree_4_intermediate_values = [E::ZERO; NUM_DEGREE_4_OPS];
+
+    let not_3 = binary_not(frame.op_bit_item(3));
+    let not_4 = binary_not(frame.op_bit_item(4));
+    let helper = frame.helper_register();
+
+    degree_4_intermediate_values[0] = not_3 * not_4;
+    degree_4_intermediate_values[1] = frame.op_bit_item(4) * not_3;
+    degree_4_intermediate_values[2] = not_4 * frame.op_bit_item(3);
+    degree_4_intermediate_values[3] = frame.op_bit_item(3) * frame.op_bit_item(4);
+    degree_4_intermediate_values.copy_within(0..4, 4);
+
+    let not_2 = binary_not(frame.op_bit_item(2));
+    for i in 0..4 {
+        degree_4_intermediate_values[i] *= not_2;
+        degree_6_intermediate_values[2 * i] = degree_4_intermediate_values[i];
+    }
+    for i in 4..8 {
+        degree_4_intermediate_values[i] *= frame.op_bit_item(2);
+        degree_6_intermediate_values[2 * i] = degree_4_intermediate_values[i];
+    }
+
+    // helper register is multiplied with all the values to enumerate all the possible
+    // degree 4 operations.
+    degree_4_intermediate_values
+        .iter_mut()
+        .take(8)
+        .for_each(|v| *v *= helper);
+
+    // The degree 6 flag (`10xxxxx`) is clubbed with both 6th bit and it's boolean not value to
+    // reduce the number of multiplications.
+    let not_5 = binary_not(frame.op_bit_item(5));
+    let degree_six_flag = frame.op_bit_item(0) * binary_not(frame.op_bit_item(1));
+    let degree_six_flag_not_5 = degree_six_flag * not_5;
+    let degree_six_flag_yes_5 = degree_six_flag * frame.op_bit_item(5);
+    for i in (0..16).step_by(2) {
+        degree_6_intermediate_values[i + 1] =
+            degree_6_intermediate_values[i] * degree_six_flag_yes_5;
+        degree_6_intermediate_values[i] *= degree_six_flag_not_5;
+    }
+
+    (degree_6_intermediate_values, degree_4_intermediate_values)
+}
+
+// TESTS
+// ================================================================================================
+#[cfg(test)]
+mod tests {
+    use crate::stack::stack_flag::StackOperationsFlag;
+
+    use super::EvaluationFrame;
+    use vm_core::{
+        decoder::{OP_BITS_OFFSET, OP_BIT_EXTRA_COL_IDX},
+        Felt, DECODER_TRACE_OFFSET, ONE, TRACE_WIDTH, ZERO,
+    };
+
+    #[test]
+    fn op_flags() {
+        // Initialize the rows.
+        let current = vec![ZERO; TRACE_WIDTH];
+        let next = vec![ZERO; TRACE_WIDTH];
+
+        // Generate a frame.
+        let frame = EvaluationFrame::<Felt>::from_rows(current.clone(), next.clone());
+
```

Comment:
I would probably create a helper function which takes an operation (i.e., `Operation` enum), builds the evaluation frame for that operation, and tests that all the op_flags for this operation are set correctly. Then i'd call this function for all operations.

If a single function would be difficult to put together - we could do 3 functions: one for degree 7 flags, another one for degree 6 flags, and the third one for degree 4 flags.